README.md

```diff
@@ -23,30 +23,61 @@
 
 ## Usage
 
-Local invocation:
+### `help`
 
 ```sh
-$ npx api-docs-tooling --help
+npx api-docs-tooling help [command]
 ```
 
+### `generate`
+
+Generate API documentation from Markdown files.
+
+```sh
+npx api-docs-tooling generate [options]
+```
+
+**Options:**
+
+- `-i, --input <patterns...>` Input file patterns (glob)
+- `--ignore [patterns...]` Files to ignore
+- `-o, --output <dir>` Output directory
+- `-v, --version <semver>` Target Node.js version (default: latest)
+- `-c, --changelog <url>` Changelog file or URL
+- `--git-ref <url>` Git ref/commit URL
+- `-t, --target [modes...]` Generator target(s): `json-simple`, `legacy-html`, etc.
+- `--no-lint` Skip linting before generation
+
+### `lint`
+
+Run the linter on API documentation.
+
+```sh
+npx api-docs-tooling lint [options]
+```
+
+**Options:**
+
+- `-i, --input <patterns...>` Input file patterns (glob)
+- `--ignore [patterns...]` Files to ignore
+- `--disable-rule [rules...]` Disable specific linting rules
+- `--dry-run` Run linter without applying changes
+- `-r, --reporter <reporter>` Reporter format: `console`, `github`, etc.
+
+### `interactive`
+
+Launches a fully interactive CLI prompt to guide you through all available options.
+
+```sh
+npx api-docs-tooling interactive
+```
+
+### `list`
+
+See available modules for each subsystem.
+
 ```sh
-Usage: api-docs-tooling [options]
```

Comment:
Can we still have a `-h` output on the top? :)

Reply:
I feel the `-h` output is more than sufficient, if any option is more complicated, can be described below.